### PR TITLE
same repo branch deletion after succesfull merge

### PR DESCRIPTION
**Pull request by khodzha**

### CODE REVIEW

**Comment by khodzha** [comment]
how should i test async behaviour?

**Comment by khodzha** [comment]
this is for issue #177

**Comment by notriddle** [comment]
You can poll the server until you get the result you want. Since it's all in-process and governed by the EVM's fair scheduler, it won't slow things down to poll without throttling.

**Comment by khodzha** [comment]
and how do i fix dialyzer complaining about
> The pattern 'true' can never match the type 'false'

since @delete_merged_branches set in config file to false?

**Comment by notriddle** [comment]
The config.exs files are global for the entire instance of bors-ng.

I think it would be a lot better if you pulled that setting out of the bors.toml. That way, different users of the public instance can pick their setting.

**Comment by notriddle** [comment]
That'll also fix the dialyzer warning.

File: apps/bors_worker/lib/batcher.ex [hunk 1]
```diff
@@ -337,6 +342,7 @@ defmodule BorsNG.Worker.Batcher do
     patches = batch.id
     |> Patch.all_for_batch()
     |> Repo.all()
+    Process.send_after(self(), {:maybe_delete_branch, batch.id}, 0)
```
Comment (notriddle) [inline apps/bors_worker/lib/batcher.ex]: Instead of send_after, you can just use send and leave off the 0.

**Pull request merged**